src/dialects/mssql/query-generator-typescript.ts

```diff
@@ -0,0 +1,56 @@
+import { noSchemaParameter } from '../../utils/deprecations';
+import { joinSQLFragments } from '../../utils/join-sql-fragments';
+import { AbstractQueryGenerator } from '../abstract/query-generator';
+import type { TableName } from '../abstract/query-interface';
+
+/**
+ * Temporary class to ease the TypeScript migration
+ */
+export class MsSqlQueryGeneratorTypeScript extends AbstractQueryGenerator {
+  // TODO [>7]: remove schema parameter
+  describeTableQuery(tableName: TableName, schema?: string) {
+    const table = this.extractTableDetails(tableName);
+
+    if (schema) {
+      noSchemaParameter();
+      table.schema = schema;
+    }
+
+    return joinSQLFragments([
+      'SELECT',
+      `c.COLUMN_NAME AS 'Name',`,
+      `c.DATA_TYPE AS 'Type',`,
+      `c.CHARACTER_MAXIMUM_LENGTH AS 'Length',`,
+      `c.IS_NULLABLE as 'IsNull',`,
+      `COLUMN_DEFAULT AS 'Default',`,
+      `pk.CONSTRAINT_TYPE AS 'Constraint',`,
+      `COLUMNPROPERTY(OBJECT_ID(c.TABLE_SCHEMA+'.'+c.TABLE_NAME), c.COLUMN_NAME, 'IsIdentity') as 'IsIdentity',`,
+      `CAST(prop.value AS NVARCHAR) AS 'Comment'`,
+      'FROM',
+      'INFORMATION_SCHEMA.TABLES t',
+      'INNER JOIN',
+      'INFORMATION_SCHEMA.COLUMNS c ON t.TABLE_NAME = c.TABLE_NAME AND t.TABLE_SCHEMA = c.TABLE_SCHEMA',
+      'LEFT JOIN (SELECT tc.table_schema, tc.table_name,',
+      'cu.column_name, tc.CONSTRAINT_TYPE',
+      'FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS tc',
+      'JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE  cu',
+      'ON tc.table_schema=cu.table_schema and tc.table_name=cu.table_name',
+      'and tc.constraint_name=cu.constraint_name',
+      `and tc.CONSTRAINT_TYPE='PRIMARY KEY') pk`,
+      'ON pk.table_schema=c.table_schema',
+      'AND pk.table_name=c.table_name',
+      'AND pk.column_name=c.column_name',
+      'INNER JOIN sys.columns AS sc',
+      `ON sc.object_id = object_id(t.table_schema + '.' + t.table_name) AND sc.name = c.column_name`,
+      'LEFT JOIN sys.extended_properties prop ON prop.major_id = sc.object_id',
+      'AND prop.minor_id = sc.column_id',
+      `AND prop.name = 'MS_Description'`,
+      `WHERE t.TABLE_NAME = ${this.escape(table.tableName)}`,
+      table.schema !== this.dialect.getDefaultSchema() ? `AND t.TABLE_SCHEMA = ${this.escape(table.schema)}` : '',
```

Comment:
IMO the schema filter should always be specified, otherwise it's going to list identically named tables from all schemas

Reply:
Fixed for the dialects that do not use `quoteTable`